sdk/Pulumi/Core/Output.cs

```diff
@@ -499,11 +499,33 @@ internal static Output<T> CreateUnknown(T value)
         internal static Output<T> CreateUnknown(Func<Task<T>> valueFactory)
             => Unknown(default!).Apply(_ => valueFactory());
 
+        /// <summary>
+        /// <see cref="Output{T}.Apply{U}(Func{T, Output{U}})"/> for more details.
+        /// </summary>
+        public Output<Task> Apply(Func<T, Task> func)
```

Comment:
Oh this is slightly confusing isn't it. How I wish we had unit. The Task returned by the Output isn't the Task returned by the func, but a different CompletedTask. Which doesn't quite align with what `Apply(Func<Task<T>>)` does (the output returns the inner T).

I have seen the suggestion of using `System.ValueTuple` as a unit type for C#, maybe we should do that here?

Reply:
> The Task returned by the Output isn't the Task returned by the func, but a different CompletedTask.

Yeah not sure why I didn't return the awaited underlying task right away 😅 I guess a bit tired. Changed that ✅ 

> I have seen the suggestion of using System.ValueTuple as a unit type for C#, maybe we should do that here?

You mean the signature would become this?
```cs
Output<Task<ValueTuple> Apply(Func<T, Task> func)
```

Reply:
No I mean it would be `Output<ValueTuple> Apply(Func<T, Task> func)`, a poor mans `unit` basically. Imagine if we're in F# land we'd just have `Output<U> Apply<U>(Func<T, Task<U>> func)` and filled in for unit you'd get `Output<unit> Apply(Func<T, unit> func)`.

This would line up with how the `Task<T>` apply functions don't return the task, but the value inside the task. Just awkward with this one because there is no value, but unit doesn't exist.

Reply:
I actually like that a lot! Having `Output<ValueTuple>` also means we could even export this thing as an output, it seems that our serialization logic doesn't handle this though, would be nice to make it output `()`